File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    
    let i = 0; // one pointer to the starting of the array

    for(let j = 0; j< nums.length;j++){
        if(nums[j] > nums[i]){
            i=i+1;
            nums[i] = nums[j]; // putting the unique elements
        }
    }

    return i+1;
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int low = 0;
        int high = low + 1;
        int length = nums.length;

        while(high < length){
            if(nums[high] == nums[high - 1]){
                high++;
                continue;
            }else{
              nums[low+1] = nums[high];
              high++;
              low++;
            }
        }

        return low+1;
        
    }
}